### PR TITLE
fix: categorizing LUIS and QNA resources correctly as required or optional in provision flow

### LINKED ISSUE #6775: QnA Maker and LUIS should be listed in required resources if needed in Publishing Profile
## Is your feature request related to a problem? Please describe.

Today, we identify and help you provision LUIS and QnA Maker when you open your bot project, however when you have publishing profiles already created and / or you create a new one you should be told that these are required resources on the Add resources modal.
image

## Describe the solution you'd like

By intelligently adding the required resources based off of your project, this will make it where you don't get an error message when you go to publish.

## Describe alternatives you've considered

Using the Next Steps menu to provide details that you should add the required resources to your publishing profiles to ensure that you give the user a heads up that all publishing profiles will now require this to be added and walk them through getting this done.  Maybe even disable publishing until these issues are addressed.

### PR DESCRIPTION
**Pull request by pavolum**
## Description

AzurePublish extension was always adding luis and qna as optional resources instead of required resources in the provision flow regardless of the underlying bot needing LUIS or QNA.

There is already a selector in client recoil state that does the parsing to determine whether LUIS/QNA are required in the current bot files (`allRequiredRecognizersSelector`)

This PR adds an accessor function for `allRequiredRecognizersSelector` that can be called through the Publish Plugin API such that the plug in can correctly categorize whether or not LUIS and QNA are optional.

## Task Item

fixes #6775 

## Screenshots

image

### CODE REVIEW

File: Composer/packages/extension-client/src/hooks/usePublishApi.ts [hunk 1]
```diff
@@ -58,6 +58,13 @@ export function usePublishApi() {
   function userShouldProvideTokens(): boolean {
     return window[ComposerGlobalName].userShouldProvideTokens();
   }
+  function getRequiredRecognizers(): {
+    projectId: string;
+    requiresLUIS: boolean;
+    requiresQNA: boolean;
+  }[] {
+    return window[ComposerGlobalName].getRequiredRecognizers();
```
Comment (benbrown) [inline Composer/packages/extension-client/src/hooks/usePublishApi.ts:66]: doesn't this need to pass the parameters through

Comment (benbrown) [reply Composer/packages/extension-client/src/hooks/usePublishApi.ts:66]: oh nevermind i see this is the return type.

File: extensions/azurePublish/src/components/azureProvisionDialog.tsx [hunk 2]
```diff
@@ -601,7 +603,9 @@ export const AzureProvisionDialog: React.FC = () => {
 
   const getResources = async () => {
     try {
-      const resources = await getResourceList(currentProjectId, publishType);
+      const requireLUIS = requiredRecognizers.some((p) => p.requiresLUIS);
```
Comment (benbrown) [inline extensions/azurePublish/src/components/azureProvisionDialog.tsx:606]: why not move this into getResourceList instead of passing it as a parameter?

**Review by benbrown — APPROVED** [review]
As discussed, I would leave the optional resources in the list from the server, and just flip them to required on the front end.

File: Composer/packages/extension-client/src/hooks/usePublishApi.ts (hunk 1 above)
Comment (pavolum) [reply Composer/packages/extension-client/src/hooks/usePublishApi.ts:66]: That function doesnt take params the bit of code 
```

File: extensions/azurePublish/src/components/azureProvisionDialog.tsx (hunk 2 above)
Comment (pavolum) [reply extensions/azurePublish/src/components/azureProvisionDialog.tsx]: Gotcha

**Pull request merged**